src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -0,0 +1,537 @@
+                break
+
+        return resource_exists
+
+    def validate_and_deploy_arm_template(
+        self, template: Any, parameters: Dict[Any, Any], resource_group: str
+    ) -> Any:
+        """
+        Validate and deploy an individual ARM template.
+
+        This ARM template will be created in the resource group passed in.
+
+        :param template: The JSON contents of the template to deploy
+        :param parameters: The JSON contents of the parameters file
+        :param resource_group: The name of the resource group that has been deployed
+                :raise RuntimeError if validation or deploy fails
+        :return: Output dictionary from the bicep template.
+        """
+        # Get current time from the time module and remove all digits after the decimal point
+        current_time = str(time.time()).split(".")[0]  # pylint: disable=use-maxsplit-arg
+
+        # Add a timestamp to the deployment name to ensure it is unique
+        deployment_name = f"AOSM_CLI_deployment_into_{resource_group}_{current_time}"
+
+        validation = self.api_clients.resource_client.deployments.begin_validate(
+            resource_group_name=resource_group,
+            deployment_name=deployment_name,
+            parameters={
+                "properties": {
+                    "mode": "Incremental",
+                    "template": template,
+                    "parameters": parameters,
+                }
+            },
+        )
+
+        validation_res = validation.result()
+        logger.debug("Validation Result %s", validation_res)
+        if validation_res.error:
+            # Validation failed so don't even try to deploy
+            logger.error(
+                "Template for resource group %s has failed validation. The message was: %s.\
+                See logs for additional details.",
+                resource_group,
+                validation_res.error.message,
+            )
+            logger.debug(
+                "Template for resource group %s failed validation. \
+                Full error details: %s",
+                resource_group,
+                validation_res.error,
+            )
+            raise RuntimeError("Azure template validation failed.")
+
+        # Validation succeeded so proceed with deployment
+        logger.debug("Successfully validated resources for %s", resource_group)
+
+        poller = self.api_clients.resource_client.deployments.begin_create_or_update(
+            resource_group_name=resource_group,
+            deployment_name=deployment_name,
+            parameters={
+                "properties": {
+                    "mode": "Incremental",
+                    "template": template,
+                    "parameters": parameters,
+                }
+            },
+        )
+        logger.debug(poller)
+
+        # Wait for the deployment to complete and get the outputs
+        deployment: DeploymentExtended = poller.result()
```

Comment:
Please use `LongRunningOperation` instead of `.result()`, which can support the progress bar by default for the long polling operations
```suggestion
        # Wait for the deployment to complete and get the outputs
        deployment: DeploymentExtended = LongRunningOperation(cmd.cli_ctx)(validation_poller)
```

Reply:
I see many similar issues elsewhere, so I won't repeat my comments

Reply:
I have changed them all